packages/salesforce/src/Adaptor.js

```diff
@@ -646,69 +646,90 @@ export function reference(position) {
   return state => state.references[position].id;
 }
 
-function setApiVersion(apiVersion) {
+function setApiVersion(state, connectionOptions) {
+  const { apiVersion } = state.configuration;
+
   const apiVersionRegex = /^\d{2}\.\d$/;
-  let version = '52.0';
+
   if (apiVersion && apiVersionRegex.test(apiVersion)) {
     console.log('Using Salesforce API version', apiVersion);
-    version = apiVersion;
+    return { ...connectionOptions, version: apiVersion };
   } else {
-    console.log('Invalid salesforce apiVersion', apiVersion);
-    console.log('Using Salesforce API version', version);
+    console.log('apiVersion is not defined');
+    console.log('We recommend using Salesforce API version 52.0 or latest');
+    return connectionOptions;
   }
+}
+
+/**
+ * Establish a connection to Salesforce using username and password.
+ * @function createBasicAuthConnection
+ * @private
+ * @param {State} state - Runtime state.
+ * @returns {State}
+ */
+async function createBasicAuthConnection(state) {
+  const { loginUrl, username, password, securityToken } = state.configuration;
 
-  return version;
+  const connectionOptions = setApiVersion(state, { loginUrl });
+
+  const connection = new jsforce.Connection(connectionOptions);
+
+  await connection
+    .login(username, securityToken ? password + securityToken : password)
+    .catch(e => {
+      console.error(`Failed to connect to salesforce as ${username}`);
+      throw e;
+    });
+
+  console.info(`Connected to salesforce as ${username}.`);
+
+  return {
+    ...state,
+    connection,
+  };
 }
+
 /**
- * Creates a connection.
- * @example
- * createConnection(state)
- * @function
+ * Establish a connection to Salesforce using Access Token.
+ * @function createAccessTokenConnection
+ * @private
  * @param {State} state - Runtime state.
  * @returns {State}
  */
-function createConnection(state) {
-  const { loginUrl, apiVersion } = state.configuration;
 
-  if (!loginUrl) {
-    throw new Error('loginUrl missing from configuration.');
-  }
+function createAccessTokenConnection(state) {
+  const { other_params, access_token } = state.configuration;
+  const { instance_url } = other_params;
+
+  const connectionOptions = setApiVersion(state, {
```

Comment:
Hmm. I'm sorry but this is still just a bit too quirky. I feel like we're making an easy thing very difficult.

How about we do this?

```
function getConnection(state, options) {
	const { apiVersion } = state.configuration;
	
	const apiVersionRegex = /^\d{2}\.\d$/;

 	if (apiVersion && apiVersionRegex.test(apiVersion)) {
    	console.log('Using Salesforce API version', apiVersion);
    	options.version = apiVersion;
    } else {
     	console.log('apiVersion is not defined');
    	console.log('We recommend using Salesforce API version 52.0 or latest');
    }

	return new jsforce.Connection(options);
 }
 

 async function createBasicAuthConnection(state) {
  const { loginUrl, username, password, securityToken } = state.configuration;

  const connection = getConnection(state, { loginUrl });

  await connection
    .login(username, securityToken ? password + securityToken : password)
    .catch(e => {
      console.error(`Failed to connect to salesforce as ${username}`);
      throw e;
    });

  console.info(`Connected to salesforce as ${username}.`);

  return {
    ...state,
    connection,
  };
}

function createAccessTokenConnection(state) {
  const { other_params, access_token } = state.configuration;
  const { instance_url } = other_params;

  const connection = getConnection(state, {
    instanceUrl: instance_url,
    accessToken: access_token,
  });

  console.log(`Connected with ${connection._sessionType} session type`);

  return {
    ...state,
    connection,
  };
}
```